reactor-netty-core/src/main/java/reactor/netty/resources/ConnectionProvider.java

```diff
@@ -488,6 +492,18 @@ public final Builder forRemoteHost(SocketAddress remoteHost, Consumer<HostSpecif
 			return this;
 		}
 
+		/**
+		 * Specifies the maximum number of connection pools that the provider can create.
+		 * If the number of connection pools created exceeds this value, a warning message is logged.
+		 * The value must be positive or zero; otherwise, the connection pools check is ignored.
+		 *
+		 * @param maxConnectionPools the number of connection pools expected to be created.
+		 * @return the current {@link Builder} instance with the updated configuration.
+		 */
+		public Builder maxConnectionPools(int maxConnectionPools) {
+			this.maxConnectionPools = maxConnectionPools;
```

Comment:
Please add a check for the provided value.

Reply:
Do we want the user to be able to reset this to `-1`? For example:

https://github.com/reactor/reactor-netty/blob/11f91b04c960286fffba0ab90e56e91440350f67/reactor-netty-core/src/main/java/reactor/netty/resources/ConnectionProvider.java#L585-L602